File: backend/app/schemas/battery.py
from datetime import datetime
from uuid import UUID

from pydantic import BaseModel


class BatteryEstimateOut(BaseModel):
    id: UUID
    vehicle_id: UUID
    calculated_at: datetime
    estimated_usable_kwh: float
    nominal_kwh: float | None
    degradation_pct: float | None
    confidence: str
    sessions_used: int
    explanation: list[str] | None

    model_config = {"from_attributes": True}


class AlertOut(BaseModel):
    id: UUID
    vehicle_id: UUID
    detected_at: datetime
    resolved_at: datetime | None
    alert_type: str
    severity: str
    confidence: str
    status: str
    title: str
    summary: str
    evidence: list[str] | None
    possible_causes: list[str] | None
    recommended_actions: list[str] | None

    model_config = {"from_attributes": True}


class DriveSessionOut(BaseModel):
    """Derived from snapshot windows where vehicle was moving."""
    started_at: datetime
    ended_at: datetime
    distance_km: float | None
    energy_kwh: float | None
    efficiency_wh_per_km: float | None
    start_soc: float | None
    end_soc: float | None
    max_speed_kmh: float | None
    duration_minutes: float


class CostSummaryOut(BaseModel):
    period: str              # e.g. "2026-05"
    total_cost: float
    total_kwh: float
    currency: str
    session_count: int
    avg_cost_per_kwh: float | None
    avg_cost_per_100km: float | None
    solar_kwh: float
    grid_kwh: float
    solar_pct: float

File: frontend/src/types/battery.ts
export interface BatteryEstimate {
  id: string;
  vehicle_id: string;
  calculated_at: string;
  estimated_usable_kwh: number;
  nominal_kwh: number | null;
  degradation_pct: number | null;
  confidence: string;
  sessions_used: number;
  explanation: string[] | null;
}

export interface DriveSession {
  started_at: string;
  ended_at: string;
  distance_km: number | null;
  energy_kwh: number | null;
  efficiency_wh_per_km: number | null;
  start_soc: number | null;
  end_soc: number | null;
  max_speed_kmh: number | null;
  duration_minutes: number;
}

export interface CostSummary {
  period: string;
  total_cost: number;
  total_kwh: number;
  currency: string;
  session_count: number;
  avg_cost_per_kwh: number | null;
  avg_cost_per_100km: number | null;
  solar_kwh: number;
  grid_kwh: number;
  solar_pct: number;
}

export interface Alert {
  id: string;
  vehicle_id: string;
  detected_at: string;
  resolved_at: string | null;
  alert_type: string;
  severity: string;
  confidence: string;
  status: string;
  title: string;
  summary: string;
  evidence: string[] | null;
  possible_causes: string[] | null;
  recommended_actions: string[] | null;
}
